Revit_Core_Engine/Convert/Environment/FromRevit/Profile.cs

```diff
@@ -0,0 +1,181 @@
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using Autodesk.Revit.DB.Structure;
+using BH.Engine.Adapters.Revit;
+using BH.Engine.Geometry;
+using BH.oM.Adapters.Revit.Settings;
+using BH.oM.Adapters.Revit.Enums;
+using BH.oM.Base;
+using BH.oM.Geometry.ShapeProfiles;
+using BH.oM.MEP.SectionProperties;
+using BH.oM.MEP.Elements;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Structure.Elements;
+using BH.oM.Structure.MaterialFragments;
+using BH.oM.Structure.SectionProperties;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using BH.oM.Geometry;
+using System.Linq;
+using BH.oM.MEP.MaterialFragments;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Convert
+    {
+        /***************************************************/
+        /****               Public Methods              ****/
+        /***************************************************/
+
+        [Description("Get the profile of a duct.")]
+        [Input("Autodesk.Revit.DB.Mechanical.DuctType", "Revit duct type.")]
+        [Input("Autodesk.Revit.DB.Mechanical.Duct", "Revit duct.")]
+        [Input("BH.oM.Adapters.Revit.Settings.RevitSettings", "Revit settings.")]
+        [Input("Dictionary<string, List<IBHoMObject>>", "Referenced objects.")]
+        [Output("IProfile", "BHoM duct section property.")]
+        public static IProfile ProfileFromRevit(this Autodesk.Revit.DB.Mechanical.DuctType ductType, Autodesk.Revit.DB.Mechanical.Duct duct, RevitSettings settings = null, Dictionary<string, List<IBHoMObject>> refObjects = null)
+        {
+            settings = settings.DefaultIfNull();
+            Options options = new Options();
+            options.IncludeNonVisibleObjects = false;
+
+            // Ensure that the duct shape is specified
+            if (ductType == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Unable to determine the duct shape.");
+
+                return null;
+            }
+
+            //List<ICurve> edges = duct.Curves(options, settings, true).FromRevit();
+            List<ICurve> edges = new List<ICurve>();
+
+            // Is the duct circular, rectangular or oval?
+            // Get the duct shape, which is either circular, rectangular, oval or null
+            Autodesk.Revit.DB.ConnectorProfileType ductShape = BH.Revit.Engine.Core.Query.DuctShape(duct, settings);
```

Comment:
Code simplified to not rely on this method

Reply:
After testing, it turns out that Revit 2018 doesn't support `ductType.Shape`, but it is supported in 2019 onwards.
Would you like me to restore my custom query method that works with all versions of Revit?

Reply:
Good point. However, string matching is a bit risky as it introduces strict language limitation, therefore better to avoid. I can see that 2018 supports `Shape` property on connectors, so you could take the first connector with `connector.ConnectorType == End` and extract the shape from it - such approach should be version agnostic, at least from the first look at the API.

Reply:
On Friday's review the custom query method has been temporarily restored until a solution based on duct connectors is successfully implemented